packages/manager/apps/web/client/app/hosting/cdn/shared/hosting-cdn-shared-settings.service.js

```diff
@@ -0,0 +1,305 @@
+      {
+        rootPath: 'apiv6',
+      },
+    );
+  }
+
+  /**
+   * List all options for a domain
+   * @param {string} serviceName: The internal name of your hosting
+   * @param {string} domainName: Domain for which the details is required
+   * @returns {*}
+   */
+  getCDNDomainsDetails(serviceName, domainName) {
+    return this.OvhHttp.get(
+      `/hosting/web/${serviceName}/cdn/domain/${domainName}/option`,
+      {
+        rootPath: 'apiv6',
+      },
+    );
+  }
+
+  /**
+   * List all options for a domain
+   * @param {string} serviceName: The internal name of your hosting
+   * @param {string} domainName: Domain for which the details is required
+   * @param {object} data: data to send
+   * @returns {*}
+   */
+  addNewOptionToDomain(serviceName, domainName, data) {
+    return this.OvhHttp.post(
+      `/hosting/web/${serviceName}/cdn/domain/${domainName}/option`,
+      {
+        rootPath: 'apiv6',
+        data,
+      },
+    );
+  }
+
+  /**
+   * Reset an option to his default value
+   * @param {string} serviceName: The internal name of your hosting
+   * @param {string} domainName: Domain for which the details is required
+   * @param {string} optionName: Option to reset to default
+   * @returns {*}
+   */
+  resetCDNOptionToDefault(serviceName, domainName, optionName) {
+    return this.OvhHttp.delete(
+      `/hosting/web/${serviceName}/cdn/domain/${domainName}/option/${optionName}`,
+      {
+        rootPath: 'apiv6',
+        data: {},
+      },
+    );
+  }
+
+  /**
+   * Get details for an option on a domain
+   * @param {string} serviceName: The internal name of your hosting
+   * @param {string} domainName: Domain for which the details is required
+   * @param {string} optionName: Option to reset to default
+   * @returns {*}
+   */
+  getCDNDomainOptionDetails(serviceName, domainName, optionName) {
+    return this.OvhHttp.get(
+      `/hosting/web/${serviceName}/cdn/domain/${domainName}/option/${optionName}`,
+      {
+        rootPath: 'apiv6',
+      },
+    );
+  }
+
+  /**
+   * Update an option on a domain
+   * @param {string} serviceName: The internal name of your hosting
+   * @param {string} domainName: Domain for which the details is required
+   * @param {string} optionName: Option to reset to default
+   * @param {object} data: data to send
+   * @returns {*}
+   */
+  updateCDNDomainOption(serviceName, domainName, optionName, data) {
+    return this.OvhHttp.put(
+      `/hosting/web/${serviceName}/cdn/domain/${domainName}/option/${optionName}`,
+      {
+        rootPath: 'apiv6',
+        data,
+      },
+    );
+  }
+
+  /**
+   * Update an option on a domain
+   * @param {string} serviceName: The internal name of your hosting
+   * @param {string} domainName: Domain for which the details is required
+   * @param {string} optionName: Option to reset to default
+   * @returns {*}
+   */
+  deleteCDNDomainOption(serviceName, domainName, optionName) {
+    return this.$http.delete(
```

Comment:
Sometimes you are using `OvhHttp` and sometimes `$http`

Reply:
Yes unfortunately for unkwon raison OvhHttp doesn't works for two endpoints.
So, i use the $http instead.